v2/app/collegevine/App.jsx

```diff
@@ -0,0 +1,1357 @@
+      n = facetsArr.length
+    for (i = 0; i < n; ++i) {
+      name = facetsArr[i]
+      facet = facets[name]
+      facetHeading = makeElement("div", "pivot pivot_facetname", filterPane)
+      facetHeading.onclick = onFacetClicked
+      facetHeading.name = name
+      facetHeading.facetType = facet.type
+      if (
+        facet.type === "String" ||
+        facet.type === "LongString" ||
+        facet.type === "Link"
+      ) {
+        // set up a few variables so we can keep track of our sorting order
+        var comparatorNames = (facetHeading.comparatorNames = [])
+        var comparators = (facetHeading.comparators = [])
+        facetHeading.currentComparator = 0
+        if (facet.orders && facet.orders.length && facet.comparator) {
+          // we'll only support one custom sort order for now, because that's what pivot
+          // seems to do. if we need to support more, we could.
+          ;(function() {
+            var temp = facet.comparator
+            comparators.push(function(a, b) {
+              return temp(a.value, b.value)
+            })
+          })()
+          comparatorNames.push("Sort: " + facet.orders[0].name)
+        }
+        comparatorNames.push("Sort: Quantity")
+        comparators.push(compareByQuantity)
+        comparatorNames.push("Sort: A-Z")
+        comparators.push(compareAlphabetical)
+      }
+      clearButton = clearButtons[name] = clearButton.cloneNode(true)
+      clearButton.onclick = onClear
+      facetHeading.appendChild(clearButton)
+      facetTitle = makeElement("div", "pivot_facetlabel", facetHeading)
+      addText(facetTitle, name)
+      facetVerticalSpace -= facetHeading.offsetHeight
+      facetOptions = makeElement("div", "pivot pivot_facetvalues", filterPane)
+      facetOptions.style.height = 0
+      facetOptions.style.overflow = "hidden"
+    }
+
+    // apply filters from URL
+    if (useHistory) {
+      hasStateBeenAppliedFromURL = true
+      history.replaceState(null, "", getPathWithState())
+      applyStateFromURL()
+    }
+  })
+
+  // any time the user interacts with the viewer, focus the offscreen text box so we can catch directional arrows
+  div.addEventListener("click", function(e) {
+    var target = e.target
+    if (target !== searchBox && target !== sortBox) {
+      inputElmt.focus()
+    }
+  })
+
+  // put the current filter state in the query after any rearrange operation
+  if (useHistory) {
+    viewer.addListener("finishedRearrange", function() {
+      if (hasStateBeenAppliedFromURL) {
+        hasStateBeenAppliedFromURL = false
+        return
+      }
+      history.pushState(null, "", getPathWithState())
+    })
+
+    window.addEventListener("popstate", function(event) {
+      hasStateBeenAppliedFromURL = true
+      applyStateFromURL()
+    })
+  }
+
+  // FIXME: Refine React POC:
+  const GridButton = props => {
+    return <button>{props.label}</button>
+  }
+  ReactDOM.render(<GridButton label="Hello World" />, topBarContainer)
```

Comment:
POC of how we can use React to enhance/rebuild parts of the UI instead of doing it the pre-2014 way (see `makeElement`).